server.js

```diff
@@ -0,0 +1,173 @@
+passport.serializeUser((user, done) => {
+  done(null, user.id);
+});
+
+// used to deserialize the user
+passport.deserializeUser((id, done) => {
+  Users.findById(id, (err, user) => {
+    done(err, user);
+  });
+});
+
+passport.use(new GoogleStrategy(
+  configAuth.googleAuth
+  , (token, refreshToken, profile, done) => {
+  // make the code asynchronous
+  // User.findOne won't fire until we have all our data back from Google
+    process.nextTick(() => {
+    // try to find the user based on their google id
+      Users.findOne({
+        where: { email: profile.emails[0].value },
+      })
+        .then((user) => {
+        // if a user is found, log them in
+          if (user) {
+            return done(null, user);
+          }
+
+          Users.create({
+            name: profile.displayName,
+            email: profile.emails[0].value,
+          })
+            .then(newUser => done(null, newUser))
+            .catch(err => done(new Error(err)));
+        })
+        .catch((error) => {
+          done(error, null);
+        });
+    });
+  },
+));
+
+app.use(session({
+  secret: process.env.SECRET,
+  key: process.env.KEY,
+  resave: false,
+  saveUninitialized: false,
+  cookie: {
+    expires: false,
+  }, // in milliseconds
+}));
+
+app.use(passport.initialize());
+
+app.use(passport.session());
+
+app.get(
+  '/auth/google',
+  passport.authenticate('google', { scope: ['profile', 'email'] }),
+);
+
+// the callback after google has authenticated the user
+app.get(
+  '/oauth2callback',
+  passport.authenticate('google', {
+    failureRedirect: '/',
+  }),
+  (req, res) => {
+    console.log(req.user.email, req.user.id, 'successfully');
+  },
+);
+
+app.use('/users', isLoggedIn, userRouter);
+app.use('/workoutTypes', typeRouter);
+app.use('/workouts', workoutRouter);
+
+// app.get('*', (req, res) => {
+//   res.sendFile(path.resolve(__dirname, 'src', 'index.html'));
+// })
+
+
+app.listen(port, () => {
+  console.log(`Server is running on port ${port}`);
+});
+
+const records = {};
+
+const job = new CronJob({
+  cronTime: '00 29 21 * * 6',
+  onTick: () => {
+    const today = new Date();
+    const monInMilliSec = new Date().setDate(today.getDate() - 7);
+    const monDate = new Date(monInMilliSec);
+    const fromDate = moment(monDate).format('YYYY-MM-DD HH:mm:ss');
+    const to = moment(today).format('YYYY-MM-DD HH:mm:ss');
+    Workouts.findAll({
+      where: {
+        createdAt: {
+          $between: [fromDate, to],
+        },
+      },
+      attributes: ['userId', [models.sequelize.fn('AVG', models.sequelize.col('count')), 'Average'], 'type'],
+      group: ['userId', 'type'],
+      order: [[models.sequelize.fn('AVG', models.sequelize.col('count')), 'DESC']],
+    })
+      .then((results) => {
+        records.workouts = results.map(result => result.toJSON());
+        const { workouts } = records;
+        const weekData = [];
+
+        workouts.map((workout) => {
```

Comment:
usually, we would assign map to a variable, and have a return statement for what each item in the `workouts` array should be mapped to (replaced with)

Reply:
Yes... I don't want to return an array. I think the forEach method would be more efficient in this case. I just want to iterate through the list and do something.